common/src/test/java/gov/cms/ab2d/common/service/JobServiceTest.java

```diff
@@ -164,11 +173,120 @@ public void updateJob() {
+
+        Resource resource = jobService.getResourceForJob(updatedJob.getJobID(), "test.ndjson");
+        Assert.assertEquals("test.ndjson", resource.getFilename());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void getFileDownloadUrlWithWrongFilename() throws IOException {
+        Job job = jobService.createJob("ExplanationOfBenefits", "http://localhost:8080");
+        OffsetDateTime now = OffsetDateTime.now();
+        OffsetDateTime localDateTime = OffsetDateTime.now();
+        job.setProgress(100);
+        job.setLastPollTime(now);
+        job.setStatus(JobStatus.IN_PROGRESS);
+        job.setCreatedAt(localDateTime);
+        job.setCompletedAt(localDateTime);
+        job.setJobID("abc");
+        job.setResourceTypes("ExplanationOfBenefits");
+        job.setRequestURL("http://localhost");
+        job.setStatusMessage("Pending");
+        job.setExpires(now);
+
+        JobOutput jobOutput = new JobOutput();
+        jobOutput.setError(false);
+        jobOutput.setFhirResourceType("ExplanationOfBenefits");
+        jobOutput.setFilePath("file.ndjson");
+        jobOutput.setJob(job);
+
+        JobOutput errorJobOutput = new JobOutput();
+        errorJobOutput.setError(true);
+        errorJobOutput.setFhirResourceType(OPERATION_OUTCOME);
+        errorJobOutput.setFilePath("error.ndjson");
+        errorJobOutput.setJob(job);
+
+        List output = List.of(jobOutput, errorJobOutput);
+        job.setJobOutput(output);
+
+        Job updatedJob = jobService.updateJob(job);
+
+        jobService.getResourceForJob(updatedJob.getJobID(), "filenamewrong.ndjson");
+    }
+
+    @Test(expected = JobOutputMissingException.class)
+    public void getFileDownloadUrlWitMissingOutput() throws IOException {
+        Job job = jobService.createJob("ExplanationOfBenefits", "http://localhost:8080");
+        OffsetDateTime now = OffsetDateTime.now();
+        OffsetDateTime localDateTime = OffsetDateTime.now();
+        job.setProgress(100);
+        job.setLastPollTime(now);
+        job.setStatus(JobStatus.IN_PROGRESS);
+        job.setCreatedAt(localDateTime);
+        job.setCompletedAt(localDateTime);
+        job.setJobID("abc");
+        job.setResourceTypes("ExplanationOfBenefits");
+        job.setRequestURL("http://localhost");
+        job.setStatusMessage("Pending");
+        job.setExpires(now);
+
+        JobOutput jobOutput = new JobOutput();
+        jobOutput.setError(false);
+        jobOutput.setFhirResourceType("ExplanationOfBenefits");
+        jobOutput.setFilePath("outputmissing.ndjson");
+        jobOutput.setJob(job);
+
+        JobOutput errorJobOutput = new JobOutput();
+        errorJobOutput.setError(true);
+        errorJobOutput.setFhirResourceType(OPERATION_OUTCOME);
+        errorJobOutput.setFilePath("error.ndjson");
+        errorJobOutput.setJob(job);
+
+        List output = List.of(jobOutput, errorJobOutput);
+        job.setJobOutput(output);
```

Comment:
We should try and keep our tests clean, too. These lines duplicate the previous test case. Let's extract them into a common private method within this test class.